src/Nethermind/Nethermind.Evm/CodeAnalysis/JumpDestinationAnalyzer.cs

```diff
@@ -59,58 +55,87 @@ public bool ValidateJump(int destination, bool isSubroutine)
+        /// be (n + 63) / 64, but that would cause overflow. So instead it's
+        /// rearranged to ((n - 1) / 64) + 1.
         /// Due to sign extension, we don't need to special case for n == 0, if we use
-        /// bitwise operations (since ((n - 1) >> 5) + 1 = 0).
-        /// This doesn't hold true for ((n - 1) / 32) + 1, which equals 1.
+        /// bitwise operations (since ((n - 1) >> 6) + 1 = 0).
+        /// This doesn't hold true for ((n - 1) / 64) + 1, which equals 1.
         ///
         /// Usage:
         /// GetInt32ArrayLengthFromBitLength(77): returns how many ints must be
         /// allocated to store 77 bits.
         /// </summary>
         /// <param name="n"></param>
         /// <returns>how many ints are required to store n bytes</returns>
-        private static int GetInt32ArrayLengthFromBitLength(int n)
+        private static int GetInt64ArrayLengthFromBitLength(int n)
         {
-            return (int)((uint)(n - 1 + (1 << BitShiftPerInt32)) >> BitShiftPerInt32);
+            return (int)((uint)(n - 1 + (1 << BitShiftPerInt64)) >> BitShiftPerInt64);
         }
 
         /// <summary>
         /// Collects data locations in code.
         /// An unset bit means the byte is an opcode, a set bit means it's data.
         /// </summary>
-        private static int[] CreateJumpDestinationBitmap(byte[] code)
+        private static long[] CreateJumpDestinationBitmap(ReadOnlySpan<byte> code)
         {
-            int[] jumpDestBitmap = new int[GetInt32ArrayLengthFromBitLength(code.Length)];
+            long[] jumpDestBitmap = new long[GetInt64ArrayLengthFromBitLength(code.Length)];
 
             int pc = 0;
+            long flags = 0;
             while (true)
             {
-                // Since we are using a non-standard for loop here;
-                // changing to while(true) plus below if check elides
-                // the bounds check from the following code array access.
-                if ((uint)pc >= (uint)code.Length) break;
+                int move = 1;
+                if (Avx2.IsSupported && (pc & 0x1f) == 0 && pc < code.Length - Vector256<sbyte>.Count)
+                {
+                    Vector256<sbyte> data = Unsafe.As<byte, Vector256<sbyte>>(ref Unsafe.AddByteOffset(ref MemoryMarshal.GetReference(code), pc));
+                    Vector256<sbyte> compare = Avx2.CompareGreaterThan(data, Vector256.Create((sbyte)PUSHx));
+                    if (compare == default)
+                    {
+                        Vector256<sbyte> dest = Avx2.CompareEqual(data, Vector256.Create((sbyte)JUMPDEST));
+                        Vector256<sbyte> sub = Avx2.CompareEqual(data, Vector256.Create((sbyte)BEGINSUB));
+                        Vector256<sbyte> combined = Avx2.Or(dest, sub);
+                        flags |= (long)Avx2.MoveMask(combined) << (pc & 0x20);
+                        move = Vector256<sbyte>.Count;
+                        goto Next;
+                    }
```

Comment:
1. Please comment
2. Lets add Vector 128bit and 512bit paths?

Reply:
Had a look at 512bit but while most of it is there some is not implemented in .NET yet, particularly `long MoveMask(Vector512<byte> value)` so I think might need to wait till .NET9 for some of the 512 stuff 😭 https://github.com/dotnet/runtime/issues/87097